src/pykeen/hpo/hpo.py

```diff
@@ -54,20 +55,22 @@
 class Objective:
     """A dataclass containing all of the information to make an objective function."""
 
-    dataset: Union[None, str, Type[Dataset]]  # 1.
+    dataset: Union[None, str, Dataset, Type[Dataset]]  # 1.
```

Comment:
`Hint[Dataset]`?

Reply:
Nope the hint is for `Hint[X] = Union[None, str, X]` and `HintType[X] = Hint[Type[X]]`. This is the only place that accepts either. I could have written `Hint[Union[X, Type[X]]` but that would be a bit overkill for one use

Reply:
Hm, it might make sense to allow passing instances for all of them then.